zerver/views/users.py

```diff
@@ -143,7 +144,11 @@ def deactivate_user_own_backend(request: HttpRequest, user_profile: UserProfile)
     if user_profile.is_realm_owner and check_last_owner(user_profile):
         raise CannotDeactivateLastUserError(is_last_owner=True)
 
-    do_deactivate_user(user_profile, acting_user=user_profile)
+    group_setting_updates = check_group_permission_updates_for_deactivating_user(user_profile)
+
+    do_deactivate_user(
```

Comment:
Users are also deactivated from LDAP and SCIM code, so need to do that also, but would be good to get an initial review on the approach once.